README.md

```diff
@@ -169,9 +169,33 @@ It includes information on our code of conduct, the process for submitting pull
 
 Remember, every contribution counts. Let's make this project better together! 💪
 
-## Set up development environment
+## Set up Docker development environment
 
-### Developing dependencies
+Docker development environment is easy to set up. It only requires [Docker](https://docs.docker.com/engine/install/) and [Docker Compose Plugin](https://docs.docker.com/compose/install/) locally. All dependencies are installed in a container but not in the host.
+
+It also has some shortcomings. In macOS, Docker is running in a Linux VM. The fs notification doesn't work well in this case. You can't use `reflex` or `nodemon` to relaunch servers when code changes. The compiler runs pretty slow too.
+
+We recommend to use Docker development environment. If Docker environment doesn't work well, like on macOS, please use [local development environment](#set-up-local-development-environment).
+
+### Build images for development
+
+```sh
+$ docker compose -f dev-compose.yaml build dev-ui dev-api
+```
+
+### Start API and UI server with Docker Compose
+
+It may take a long time to build dependencies when launching servers first time.
+
+```sh
+$ docker compose -f dev-compose.yaml up dev-api dev-ui
+```
```

Comment:
Sorry for the back-and-forth process, but I didn't mean to replace the `docker run` commands with the single compose command, as for sure `docker run` commands are useful for certain cases. Can you please provide both ways in the doc here, so contributors can decide which is better for them?

Reply:
Added below.